controllers/configurationpolicy_controller.go

```diff
@@ -3325,12 +3325,10 @@ func (r *ConfigurationPolicyReconciler) checkAndUpdateResource(
 				diff = handleDiff(log, recordDiff, existingObjectCopy, mergedObjCopy, r.FullDiffs)
 			}
 
-			// Assume object is compliant by inverting the value of throwSpecViolation
-			r.setEvaluatedObject(obj.policy, obj.existingObj, !throwSpecViolation, "")
+			// treat the object as compliant, with no updates needed
+			r.setEvaluatedObject(obj.policy, obj.existingObj, true, "")
 
-			matchesAfterDryRun = true
-
-			return throwSpecViolation, "", diff, updateNeeded, updatedObj, matchesAfterDryRun
+			return false, "", diff, false, updatedObj, true
```

Comment:
The block above this considers `status` and this change would override the status check where it'd never consider a differing status noncompliant. Aside from re-assigning `matchesAfterDryRun`, this code block might need to stay as-is? Or is the bug with the logic behind `throwSpecViolation`?